lib/lockfile.go

```diff
@@ -0,0 +1,52 @@
+package lib
+
+import (
+	"fmt"
+	"os"
+	"time"
+)
+
+// Create exclusive lock
+func acquireLock(lockFile string, LockWaitMaxAttempts int, LockWaitInterval time.Duration) error {
+	var waitForLock bool = true
+	var LockWaitCount int = 0
+
+	logger.Debugf("Attempting to acquire lock %q", lockFile)
+	for waitForLock {
+		LockWaitCount++
+		if _, err := os.OpenFile(lockFile, os.O_CREATE|os.O_EXCL, 0o644); err != nil {
+			if LockWaitCount > LockWaitMaxAttempts {
+				return fmt.Errorf("Unable to acquire lock %q", lockFile)
```

Comment:
In these logs, do we wish to refer to it as a lock file?
```suggestion
				return fmt.Errorf("Unable to acquire lock file %q", lockFile)
```

Reply:
I'd prefer it to be referred as "lock" rather than a "file" as its main idea is locking concurrent access, rather than handling a file. Does it sound reasonable?